tests/test_utils.py

```diff
@@ -179,11 +180,11 @@ def test_get_ssl_cert_file():
             {"https": "http://127.0.0.1", "http": "https://127.0.0.1"},
         ),
         (
-            {"all_proxy": "http://127.0.0.1", "ALL_PROXY": "https://1.1.1.1"},
+            {"all_proxy": "http://127.0.0.1", "_ALL_PROXY": "https://1.1.1.1"},
```

Comment:
This change isn't clear to me?

Reply:
the reason is case insensitive problem. `all_proxy` env is affected by upper case `ALL_PROXY`. Because of that, I added an underscore.

Reply:
That's the reason this test is here, to verify that lowercase is preferred over uppercase.  On Windows is that not true?

Reply:
os.environ is case-insensitive on windows os.  in `os.environ`, there is just `ALL_PROXY`
linux:
```
>>>os.environ['all_proxy']
'http://127.0.0.1'
>>>os.environ['ALL_PROXY']
'https://1.1.1.1'
```

windows:
```
>>>os.environ['all_proxy']
'https://1.1.1.1'
>>>os.environ['ALL_PROXY']
'https://1.1.1.1'
```

Reply:
We can use `nt.envrion` but have to change `urllib.request.getproxies`

Reply:
> to verify that lowercase is preferred over uppercase

Do we want that behavior in any case? I'm slightly surprised that we support lowercased envvars (on case sensitive platforms) since the convention in these cases is usually UPPERCASED_NAMES, no?

Reply:
@tomchristie This is another case of "we support it because Requests (and curl) supports it".

Reply:
Okay, that's fair enough. I don't think we *neccessarily* need to specify or test which is prefered, tho I'd probably tend to uppercase first.

(Setting both variants is a broken behaviour that's going to lead to uncertainty regardless of anything we specify)